clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -660,14 +672,14 @@ private HeartbeatRequestManager createHeartbeatRequestManager() {
         subscriptions = mock(SubscriptionState.class);
         membershipManager = mock(MembershipManager.class);
         backgroundEventHandler = mock(BackgroundEventHandler.class);
-        heartbeatState = new HeartbeatRequestManager.HeartbeatState(subscriptions, membershipManager, maxPollIntervalMs);
-        heartbeatRequestState = new HeartbeatRequestManager.HeartbeatRequestState(
+        heartbeatState = spy(new HeartbeatRequestManager.HeartbeatState(subscriptions, membershipManager, maxPollIntervalMs));
```

Comment:
the purpose of spying was to make sure a certain method was invoked. I think this is an acceptable use case, do you agree?

Reply:
No, I don't.  Why not passing mocks to this method.